lib/compiler-singlepass/src/machine.rs

```diff
@@ -498,12 +498,12 @@ impl Machine {
         n_params: u32,
         _calling_convention: CallingConvention,
     ) {
-        let locals_size = (n as usize)
-            .saturating_sub(
-                Self::LOCAL_REGISTERS
-                    .len()
-                    .saturating_sub(n_params as usize),
-            )
+        let registers_remaining_for_locals = Self::LOCAL_REGISTERS
+            .len()
+            .saturating_sub(n_params as usize);
+        let locals_to_init = (n - n_params) as usize;
+        let locals_size = locals_to_init
+            .saturating_sub(registers_remaining_for_locals)
```

Comment:
FWIW I don't believe we should be placing locals into registers at all (there's no relation whatsoever between their index and how frequently they get used), but that’s something that we should fix as a follow-up.

Reply:
Yea agreed; I think the only benefit it has is for functions with less than LOCAL_REGISTERS locals, where all the locals then fit into registers, but it’s not actually improving the worst-case and thus we may be better off with it removed (modulo benchmarking)